source/extensions/filters/network/redis_proxy/conn_pool_impl.cc

```diff
@@ -271,6 +271,7 @@ InstanceImpl::ThreadLocalPool::makeRequestToHost(const std::string& host_address
     // This host is not known to the cluster manager. Create a new host and insert it into the map.
     if (created_via_redirect_hosts_.size() == parent_.config_.maxUpstreamUnknownConnections()) {
       // Too many upstream connections to unknown hosts have been created.
+      parent_.redis_cluster_stats_.max_upstream_unknown_connections_reached_.inc();
```

Comment:
Can you add a test for this?

Reply:
yes, there is an existing test for this exit, and I've added a check of this counter's value as well.